### ISSUE TITLE
Grav 1.7 Migration Multilanguage

### ISSUE THREAD

**Issue by NicoHood** [getgrav/grav-learn#858]
https://learn.getgrav.org/17/advanced/grav-development/grav-17-upgrade-guide#multi-language
https://learn.getgrav.org/17/basics/grav-configuration#languages

I would like to have an example configuration and an example usecase.

I can imagine the following setup:
* Supported languages: DE, EN, FR
* Default language is DE

Is this a usecase where someone want to set the content_fallback to english? Like:
de -> en
en -> en (or will this fall back to de??)
fr -> en

Or is it even possible to add a detailed fallback mechanism like
de -> en
en -> de
fr -> en

It is not really clear what are the goals, the limit and the syntax of this setting (array, string, both).

**Comment by mahagr** [comment]
Basically you set list of languages and Grav displays you the content using the first language that matches the list in your selected language. If page has no matching version, 404 is displayed instead.

So:
* User has `de` selected
* Grav finds the page (default.md)
* Key `de: en` gets selected
* If list does not contain `de`, it gets added to the first item
* If list contains default language '`de`' and no '' (empty string), file without language suffix gets added after `de`
* Lookup is as follows: `de: ['de', '', 'en']` => `default.de.md`, `default.md`, `default.en.md`

There's nothing more in it.

**Comment by mahagr** [comment]
> Basically you set list of languages and Grav displays you the content using the first language that matches the list in your selected language. If page has no matching version, 404 is displayed instead.
> 
> So:
> 
>     * User has `de` selected
> 
>     * Grav finds the page (default.md)
> 
>     * Key `de: en` gets selected
> 
>     * If list does not contain `de`, it gets added to the first item
> 
>     * If list contains default language '`de`' and no '' (empty string), file without language suffix gets added after `de`
> 
>     * Lookup is as follows: `de: ['de', '', 'en']` => `default.de.md`, `default.md`, 'default.en.md`
> 
> 
> There's nothing more in it.

**Comment by mahagr** [comment]
Sorry, misclick.

**Comment by NicoHood** [comment]
I played a bit with it and I think it gets clearer now. However **I think I ran into a bug**, correct me if I am wrong:
```yaml
languages:
  translations: true
  http_accept_language: true
  include_default_lang: false
  default_lang: de
  include_default_lang_file_extension: true
  content_fallback:
    fr: [en]
    en: [de]
    de: [en]
  supported:
   - de
   - en
   - fr
```

Files on disk:
```
modular.md
modular.fr.md
modular.en.md
```

For simplicity I've only changed the `page.title` in each file. When trying to access `/fr` the page loads but has not content (the markdown stuff) at all (except the page title is set). I think it tries to load a fallback, even though there IS an existing page. It works if I use: `fr: [en, de]`

What I did not yet understand is what you mean by the `''` empty string. I could not yet find a working case where specifying an empty string would change something.

Now speaking in general, what setting would make most sense? From my understanding this is a common usecase, **is it correct?**:
```
    fr: [en, de]
    en: [de]
    de: [en]
```

**Comment by mahagr** [comment]
You're likely missing the fact that the translation should have all the fields -- no content is added from `modular.md` file. This is something I would like to fix, but we really cannot without first getting rid of the old page logic.

Ideally I would love to have an option where you would have base file and just the overrides for every other language.

**Comment by NicoHood** [comment]
I think you missunderstood, however this seems to be fixed in 1.7.0 stable release anyways.
